File: Sample-ObjC/08-fetch-n-update-data/Mesosfer-ObjC/MainViewController.m
/**
 * Copyright (c) 2016, Mesosfer
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "MainViewController.h"
#import "UIViewController+Helper.h"
#import <Mesosfer/Mesosfer.h>

@interface MainViewController ()

@end

@implementation MainViewController

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    if (indexPath.row == 3) {
        [MFUser logOutAsyncWithBlock:^(NSError * _Nullable error) {
            if (error) {
                [self showErrorWithTitle:@"Failed to Logout" error:error handler:nil];
                return;
            }
            
            // successfully logout, dismiss main view controller
            [self dismissViewControllerAnimated:YES completion:nil];
        }];
    }
}

@end

File: Sample-Swift/05-query-user/Mesosfer-Swift/MainViewController.swift
/**
 * Copyright (c) 2016, Mesosfer
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 */

import Foundation
import UIKit
import Mesosfer

class MainViewController: UITableViewController {
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        // profile
        if indexPath.row == 2 {
            MFUser.logOutAsync(block: { (error) in
                if let e = error as? NSError {
                    self.showError(title: "Failed to Logout", error: e)
                    return
                }
                
                // successfully logout, dismiss main view controller
                self.dismiss(animated: true)
            })
        }
    }
}
